app/src/main/java/com/umbrella/umbrella/Course.java

```diff
@@ -0,0 +1,118 @@
+package com.umbrella.umbrella;
+
+import java.util.LinkedList;
+
+/**
+ * Created by wauch on 2018-02-19.
+ * A course class that holds course-related information such as pre-requisites and IDs
+ */
+
+public class Course {
+
+    /*To have attributes added as needed, modify after DB integration*/
+    private String CRN;
+    private String courseID;
+    private String courseName;
+    public enum CourseType {
+    CRS, TUT, LAB
+    }
+    private CourseType courseType;
+    private String Faculty;
+    private CourseSet preReq;
+    private CourseSet coReq;
+
+    public Course(){
+        preReq = new CourseSet();
+        coReq = new CourseSet();
+        courseType = CourseType.CRS;
+
+    }
+
+    public Course(String CRN, String courseID, String courseName) {
+        preReq = new CourseSet();
+        coReq = new CourseSet();
+        this.CRN = CRN;
+        this.courseID = courseID;
+        this.courseName = courseName;
+    }
+
+    public Course(String CRN, String courseID, String courseName, String courseType) {
+        preReq = new CourseSet();
+        coReq = new CourseSet();
+        this.CRN = CRN;
+        this.courseID = courseID;
+        this.courseName = courseName;
+
+        if(courseType.equals("0") || courseType.equals("CRS") || courseType.equals("course") || courseType.equals("crs")){
+            this.courseType = CourseType.CRS;
+        }
+
+        if(courseType.equals("1") || courseType.equals("LAB") || courseType.equals("laboratory") || courseType.equals("lab")){
+            this.courseType = CourseType.LAB;
+        }
+
+        if(courseType.equals("0") || courseType.equals("TUT") || courseType.equals("tutorial") || courseType.equals("tut")){
+            this.courseType = CourseType.TUT;
+        }
+
+    }
+
+
+    public Course(String CRN, String courseID, String courseName, CourseSet preReq, CourseSet coReq) {
```

Comment:
Lots of parameters... maybe make some optional? (eg builder pattern, default to empty sets etc)